From 4028dbf168a27bef327c4119a0eabc2155bace5e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 01 Apr 2020 13:32:09 +0300
Subject: [PATCH] [SW-11180] ppboot: move preboot offset assembly

Move the Preboot offset assembly to preboot_image_offset() function
solely.
This will enable to (re)use the function with the new boot descriptors.

Change-Id: Ia6a3b76aed4efa8c76dc451de0c3eb84747ad7a4
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index 71347a9..b78dde2 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -41,7 +41,8 @@
 		name = "MODIFIABLE";
 	}
 
-	*offset = next_image_base_offset(next_img_off, name);
+	*offset = next_image_base_offset(next_img_off, name) +
+							HL_SPI_PREBOOT_OFF;
 
 	return image_pending;
 }
@@ -51,7 +52,6 @@
 	char *name = "PREBOOT";
 	u32 img_offset;
 	enum fw_post_reset_cmd upd_pending = preboot_image_offset(&img_offset);
-	u32 preboot_offset = img_offset + HL_SPI_PREBOOT_OFF;
 
 	preboot_puts_raw("Preparing to run Preboot ");
 	if (upd_pending != FLOPS_UPD_PENDING)
@@ -59,10 +59,10 @@
 	else
 		preboot_puts_raw("candidate");
 
-	preboot_puts_hex_raw(" (offset: ", preboot_offset, false);
+	preboot_puts_hex_raw(" (offset: ", img_offset, false);
 	preboot_puts(")\n");
 
-	spi_copy((void *)PREBOOT_TEXT_BASE, preboot_offset, PREBOOT_SIZE, name);
+	spi_copy((void *)PREBOOT_TEXT_BASE, img_offset, PREBOOT_SIZE, name);
 }
 
 static void rst_src_dump(void)
